spec.bs

```diff
@@ -5369,15 +5429,77 @@ result of [=evaluating a bidding script=], or an [=additional bid=] provided by
+     Requires the first that many given ad components to be included.
 </dl>
 
+<div algorithm>
+
+To <dfn>apply any component ads target to a bid</dfn>, given a [=generated bid=] |generatedBid|:
+1. If |generatedBid|'s [=generated bid/target number of ad components=] is null, return.
+1. [=Assert=] that |generatedBid|'s [=generated bid/ad component descriptors=] is not null.
+1. Truncate |generatedBid|'s [=generated bid/ad component descriptors=] to length [=generated bid/target number of ad components=].
+
+</div>
+
+<div algorithm>
+
+To <dfn>try to reach component ads target considering k-anonymity</dfn>, given a [=generated bid=] |generatedBid|:
+1. If |generatedBid|'s [=generated bid/target number of ad components=] is null, return false.
+1. Let |selectedComponents| be a new [=list=] of [=ad descriptors=].
+1. [=set/For each=] |i| of [=list/get the indices=] of |generatedBid|:
+  1. Let |candidateComponent| be |generatedBid|'s [=generated bid/ad component descriptors=][|i|].
+  1. If [=query component ad k-anonymity count=] given |candidateComponent|'s [=interest group ad/render url=] returns true:
+    1. [=list/Append=] |candidateComponent| to |selectedComponents|.
+  1. Otherwise:
+    1. If |i| &lt; |generatedBid|'s [=generated bid/number of mandatory ad components=], return false.
+1. If |selectedComponents|'s [=list/size=] &lt; |generatedBid|'s [=generated bid/target number of ad components=],
+   return false.
+1. Set |generatedBid|'s [=generated bid/ad component descriptors=] to |selectedComponents|.
+1. [=Apply any component ads target to a bid=] given |generatedBid|.
+1. Return [=query generated bid k-anonymity count=] given |generatedBid|.
+
+Issue: TODO: change to query k-anonymity cache instead, once
+(<a href="https://github.com/WICG/turtledove/pull/1005">WICG/turtledove#868</a>) is merged
+
+</div>
+
+<div algorithm>
+
+To <dfn>adjust bid list based on k-anonymity</dfn> given a [=list=] of [=generated bids=] |bidsBatch|:
+1. Let |bidsToScore| be a new [=list=] of [=generated bids=]
+1. [=list/For each=] |generatedBid| of |bidsBatch|:
+  1. Let |bidCopy| be a clone of |generatedBid|.
+  1. Set |bidCopy|'s [=generated bid/for k-anon auction=] to false.
+  1. [=Apply any component ads target to a bid=] given |bidCopy|.
+  1. [=list/Append=] |bidCopy| to |bidsToScore|
+  1. If [=query generated bid k-anonymity count=] given |generatedBid| returns true:
+    1. [=list/Append=] |generatedBid| to |bidsToScore|.
+
+    Note: Conceptually, a bid that's already k-anonymous is considered
+    for both the k-anonymous and non-enforcing-k-anonymity leadership.
+
+  1. Otherwise, if the result of applying [=try to reach component ads target considering k-anonymity=]
+     given |generatedBid| is true:
+    1. [=list/Append=] |generatedBid| to |bidsToScore|.
+1. Return |bidsToScore|.
+
+Issue: TODO: change to query k-anonymity cache instead, once

```

Comment:
similarly, you need an empty line and then an extra indent to put it next to the step you'd like the Issue to be associated with.

Reply:
Done